src/monster.cpp

```diff
@@ -479,7 +477,7 @@ bool Monster::isOpponent(const Creature* creature) const
 			return true;
 		}
 	} else {
-		if ((creature->getPlayer() && !creature->getPlayer()->hasFlag(PlayerFlag_IgnoredByMonsters)) ||
+		if ((creature->getPlayer() && !static_cast<const Player*>(creature)->hasFlag(PlayerFlag_IgnoredByMonsters)) ||
```

Comment:
I think this is better:
```suggestion
		if (Player* player = creature->getPlayer(); player && !player->hasFlag(PlayerFlag_IgnoredByMonsters)) ||
```